src/modules/kvs-watch/kvs-watch.c

```diff
@@ -228,18 +240,128 @@ static bool key_match (json_t *o, const char *key)
 static void watcher_cleanup (struct ns_monitor *nsm, struct watcher *w)
 {
     /* wait for all in flight lookups to complete before destroying watcher */
-    if (zlist_size (w->lookups) == 0)
+    if (zlist_size (w->lookups) == 0 && zlist_size (w->loads) == 0)
         zlistx_delete (nsm->watchers, w->handle);
     /* if nsm->getrootf, destroy when getroot_continuation completes */
     if (zlistx_size (nsm->watchers) == 0
         && !nsm->getrootf)
         zhash_delete (nsm->ctx->namespaces, nsm->ns_name);
 }
 
+static void handle_load_response (flux_future_t *f, struct watcher *w)
+{
+    flux_t *h = flux_future_get_flux (f);
+    const void *data;
+    int size;
+
+    if (content_load_get (f, &data, &size) < 0) {
+        flux_log_error (h, "failed to load content data");
+        goto error;
+    }
+
+    if (!w->mute) {
+        json_t *val = treeobj_create_val (data, size);
+        if (!val) {
+            flux_log_error (h, "%s: treeobj_create_val", __FUNCTION__);
+            goto error;
+        }
+        if (flux_respond_pack (h, w->request, "{ s:o }", "val", val) < 0) {
+            flux_log_error (h, "%s: flux_respond_pack", __FUNCTION__);
+            json_decref (val);
+            goto error;
+        }
```

Comment:
After the `goto error`, another error response is attempted.  I think we've settled previously that this is not a good idea.  Just log the (unlikely) error.

In new code I've been trying to log messages like "failed to respond to TOPIC" in these cases.